### PR TITLE
close gorouting when context cancel

### PR DESCRIPTION
**Pull request by qianlnk**
startreceiver with context cancel

when I cancel, I find the go routing func not exit, and it always get error "long poll request returned error".

### CODE REVIEW

File: pkg/cloudevents/transport/http/transport.go [hunk 1]
```diff
@@ -383,19 +383,26 @@ func (t *Transport) longPollStart(ctx context.Context) error {
...
 		for {
+			if isClosed {
+				return
+			}
+
 			if resp, err := t.LongPollClient.Do(req); err != nil {
 				logger.Errorw("long poll request returned error", err)
 				uErr := err.(*url.Error)
 				if uErr.Temporary() || uErr.Timeout() {
+					time.Sleep(time.Second * 2)
```
Comment (n3wscott) [inline pkg/cloudevents/transport/http/transport.go]: not sure why you would want to add sleep statements here. Is this for debugging and forgotten?

Comment (qianlnk) [reply pkg/cloudevents/transport/http/transport.go]: when get error here， it retry "DO" too many times. then you will get too many error log, so I think here should retry after some time.

Comment (n3wscott) [reply pkg/cloudevents/transport/http/transport.go]: We can't add seep commands to the codebase like this.

Comment (qianlnk) [reply pkg/cloudevents/transport/http/transport.go]: OK, I've deleted the sleep code.

**Comment by n3wscott** [comment]
LGTM

**Pull request merged**